src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,24 @@
 import React from 'react';
+import { Status } from '../../types/Status';
+import { useAppDispatch, useAppSelector } from '../../app/store';
+import { filterSlice } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+
+  const selectStatusTodosHandler = (statusValue: Status) => {
+    dispatch(filterSlice.actions.setStatus(statusValue));
+  };
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={e => selectStatusTodosHandler(e.target.value as Status)}
```

Comment:
The `selectStatusTodosHandler` function casts `e.target.value` to `Status`. Ensure that the `Status` type includes all possible values ('all', 'active', 'completed') to avoid runtime errors. Consider validating the value before casting or using a type guard.